packages/shared/src/components/post/write/CreatePostButton.tsx

```diff
@@ -43,20 +44,34 @@ export function CreatePostButton({
     return !allowedToPost;
   };
 
-  return (
+  const href =
+    link.post.create + (squad && allowedToPost ? `?sid=${squad.handle}` : '');
+
+  const renderBtn = () => (
```

Comment:
You can make it `RenderBtn` and then `<RenderBtn />` below, more react-ish.

Reply:
See https://github.com/dailydotdev/apps/pull/2722#discussion_r1511162601

Reply:
updated based on @nensidosari suggestion, looking cleaner now 🙏